components/automate-ui/src/app/entities/automate-settings/automate-settings.model.ts

```diff
@@ -1,73 +1,141 @@
+
+  // Client Runs
+  ClientRunsRemoveData = 'clientRunsRemoveData',
+  ClientRunsLabelMissing = 'clientRunsLabelMissing',
+  ClientRunsRemoveNodes = 'clientRunsRemoveNodes',
 
-  // DeleteNodes: Removes completely from elasticsearch nodes that
-  // have been marked for deletion
-  DeleteNodes = 'delete_nodes'
+  // Compliance
+  ComplianceRemoveReports = 'complianceRemoveReports',
+  ComplianceRemoveScans = 'complianceRemoveScans'
 }
 
 export class IngestJob {
-  running: boolean;
+  category: string;
   name: string;
+  nested_name?: string;
+  recurrence?: string;
   threshold: string;
-  every?: string;
-  lastRun?: Date;
-  nextRun?: Date;
-  lastElapsed?: Date;
-  startedOn?: Date;
+  disabled: boolean;
+  purge_policies?: any; // needs specific type
+  last_elapsed?: Date;
+  next_due_at?: Date;
+  last_enqueued_at?: Date;
+  last_started_at?: Date;
+  last_ended_at?: Date;
 
-  constructor(respJob: RespJob) {
+  constructor(category: string, respJob: RespJob) {
     if (respJob !== null) {
-      this.running = respJob.running;
+      this.category = category;
       this.name = respJob.name;
-      this.every = respJob.every;
+      this.nested_name = respJob.nested_name;
+      this.disabled = respJob.disabled;
+      this.recurrence = respJob.recurrence;
       this.threshold = respJob.threshold;
-      this.lastRun = new Date(respJob.last_run);
-      this.nextRun = new Date(respJob.next_run);
-      this.lastElapsed = new Date(respJob.last_elapsed);
-      this.startedOn = new Date(respJob.started_on);
+      this.purge_policies = respJob.purge_policies;
+      this.last_elapsed = new Date(respJob.last_elapsed);
+      this.next_due_at = new Date(respJob.next_due_at);
+      this.last_enqueued_at = new Date(respJob.last_enqueued_at);
+      this.last_started_at = new Date(respJob.last_started_at);
+      this.last_ended_at = new Date(respJob.last_ended_at);
     }
   }
 }
+
+export class UnfurledJob {
+  disabled: boolean;
+  policy_name?: string;
+  older_than_days?: number;
+  name?: string;
+  threshold?: string;
+}
+
+export interface JobRequestComponent {
+  name: string;
+  purge_policies: {
+    elasticsearch: UnfurledJob[];
+  };
+}
+
+export interface JobRequestBody {
+  infra: {
+    // Infra is a special case and con contain a JobRequestComponent
+    // or an UnfurledJob depending on the API
+    job_settings: JobRequestComponent[] | UnfurledJob;
+  };
+  compliance: {
+    job_settings: JobRequestComponent[]
+  };
+  event_feed: {
+    job_settings: JobRequestComponent[]
+  };
+  // services has not yet been implemented so we will leave as optional for now
+  services?: {
+    job_settings: JobRequestComponent[]
+  };
+}
+
+export enum InfraJobName {
+  MissingNodes = 'missing_nodes',
+  MissingNodesForDeletion = 'missing_nodes_for_deletion',
+  DeleteNodes = 'delete_nodes',
+  PeriodicPurgeTimeseries = 'periodic_purge_timeseries'
+}
+
+export enum NestedJobName {
```

Comment:
Might be worth a comment to the effect that jobs that are nested within the _xyz_ property of an infra job must be one of these.